packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1043,20 +1058,30 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
   }
 }
 
-class _TimePickerDialog extends StatefulWidget {
-  const _TimePickerDialog({
+/// A material design time picker designed to appear inside a popup dialog.
+///
+/// Pass this widget to [showDialog]. The value returned by [showDialog] is the
+/// selected [TimeOfDay] if the user taps the "OK" button, or null if the user
+/// taps the "CANCEL" button. The selected time is reported by calling
+/// [Navigator.pop].
+class TimePickerDialog extends StatefulWidget {
```

Comment:
do we only expose this for testing?

Reply:
seems weird to expose a TimePickerDialog but not a DatePickerDialog.
seems weird to expose either if they're an implementation detail of showTimePicker et al.

Reply:
Done (used custom Navigator/MediaQuery/Localizations/Directionality).